locksmith/src/operations/userOperations.ts

```diff
@@ -162,9 +163,10 @@ namespace UserOperations {
     const user = await UserReference.findOne({
       where: { emailAddress: Normalizer.emailAddress(emailAddress) },
     })
-
-    if (user) {
-      return getCardDetailsFromStripe(user.stripe_customer_id)
+    if (user && user.publicKey) {
+      return getCardDetailsFromStripe(
```

Comment:
It might make sense to decompose this a bit to make updating the tests for this a bit easier